### ISSUE TITLE
Regression: KeyMustBeAString errors with RLS

### ISSUE THREAD

**Issue by YaLTeR** [autozimu/LanguageClient-neovim#406]
Starting from e1b61cca77bc9b2ba1c5e7a350102f8d61df290a the language client doesn't work properly with RLS and outputs `KeyMustBeAString` errors in `/tmp/LanguageClient.log`, example output:
```
1:04:43 INFO Handler-rust src/languageclient.rs:2169 Begin window/progress
21:04:43 INFO Handler-rust src/languageclient.rs:2169 Begin window/progress
21:04:43 ERROR Handler-rust src/vim.rs:349 Error handling message: key must be a string
Message: {"jsonrpc":"2.0","method":"window/progress","params":{"id":"progress_0","title":"Indexing"}}
Error: ErrorImpl { code: KeyMustBeAString, line: 0, column: 0 }
21:04:43 ERROR Handler-rust src/vim.rs:349 Error handling message: key must be a string
Message: {"jsonrpc":"2.0","method":"window/progress","params":{"done":true,"id":"progress_1","title":"Building"}}
Error: ErrorImpl { code: KeyMustBeAString, line: 0, column: 0 }
21:04:43 INFO RPC-rust src/vim.rs:332 <= {"jsonrpc":"2.0","method":"window/progress","params":{"done":true,"id":"progress_0","title":"Indexing"}}
21:04:43 INFO Handler-rust src/languageclient.rs:2169 Begin window/progress
21:04:43 ERROR Handler-rust src/vim.rs:349 Error handling message: key must be a string
Message: {"jsonrpc":"2.0","method":"window/progress","params":{"done":true,"id":"progress_0","title":"Indexing"}}
Error: ErrorImpl { code: KeyMustBeAString, line: 0, column: 0 }
21:04:44 INFO main src/vim.rs:332 <= {"method": "exit", "jsonrpc": "2.0", "params": {"languageId": "rust", "buftype": ""}}
21:04:44 INFO Handler-main src/languageclient.rs:1735 Begin exit
21:04:44 INFO Handler-main src/languageclient.rs:46 gather_args: [LanguageId] = [String("rust")]
21:04:44 INFO Handler-main src/vim.rs:156 => {"jsonrpc":"2.0","method":"exit","params":null}
21:04:44 ERROR Handler-main src/languageclient.rs:1744 Error: ErrorImpl { code: KeyMustBeAString, line: 0, column: 0 }
21:04:44 ERROR Handler-main src/languageclient.rs:1748 Error: ErrorImpl { code: KeyMustBeAString, line: 0, column: 0 }
```
When opening files the status message gets stuck at `LS: Building...`, the functionality doesn't work and the log looks like that with errors after most messages.

**Comment by autozimu** [comment]
I cannot reproduce this locally with both RLS stable (rls-preview 0.125.1-stable (cebf188 2018-03-19)) and nightly (rls-preview 0.126.0-nightly (9144e22 2018-04-23)) using latest commit, and I don't see any suspicious from the log.

Please provide this plugin version and rls version.

**Comment by YaLTeR** [comment]
Looks like it was fixed in ddbb8fad5b812b3f92a2a67148a1a720c4cf9a3c. My bad for bisecting with an outdated HEAD.
